Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -95,6 +95,9 @@ void LogGrpcCallFinished(absl::string_view rpc_name,
       grpc_connection_{database_info, worker_queue, &grpc_queue_,
                        connectivity_monitor_.get()},
       serializer_bridge_{NOT_NULL(serializer)} {
+}
+
+void Datastore::Start() {
```

Comment:
Moving the polling of gRPC completion queue from the constructor into a method allows tests to bypass the normal polling mechanism (otherwise, it may pull off completions from the queue before `ForceFinish` can get to them).